shpc/main/modules/templates/docker.lua

```diff
@@ -25,7 +25,7 @@ Commands include:
        {{ command }} inspect <container>
 
 {% if aliases %}{% for alias in aliases %} - {{ alias.name }}:
-       {{ command }} run -i{% if tty %}t{% endif %} -u `id -u`:`id -g` --rm --entrypoint {{ alias.entrypoint }} {% if envfile %}--env-file {{ module_dir }}/{{ envfile }}{% endif %} {% if bindpaths %}-v {{ bindpaths }} {% endif %}{% if features.home %}-v {{ features.home }} {% endif %}{% if alias.docker_options %}{{ alias.docker_options }} {% endif %} -v ${PWD} -w ${PWD} <container> "{{ alias.args }}"
+       {{ command }} run -i{% if tty %}t{% endif %} -u `id -u`:`id -g` --rm --entrypoint {{ alias.entrypoint }} {% if envfile %}--env-file {{ module_dir }}/{{ envfile }}{% endif %} {% if bindpaths %}-v {{ bindpaths }} {% endif %}{% if features.home %}-v {{ features.home }} {% endif %}{% if alias.docker_options %}{{ alias.docker_options }} {% endif %} -v ${PWD} -w ${PWD} <container> "{{ alias.command }}" "$@"
```

Comment:
I'm wondering why this works - e.g.,:

- the full entrypoint + args starts as a value, this is alias.command
- we split this value into args.entrypoint and args.args (see here https://github.com/singularityhub/singularity-hpc/blob/7b844f15b89d09431e3eb5ed54abc0ee171aa92d/shpc/main/container/__init__.py#L194-L198)
- so what we are doing now is having the entrypoint PLUS repeating it for the exec command at the end. Does this really work? And to step back - why wasn't the entrypoint being honored before? I am thinking maybe we need to use alias.command for singularity, but maintain alias.entrypoint -> entrypoint and alias.args for the doker/podman templates?

Reply:
you're right, `args` is the right one!! :)